code/modules/paperwork/ticketmachine.dm

```diff
@@ -0,0 +1,201 @@
+	var/id = 1
+
+/obj/machinery/ticket_machine/emag_act(mob/user) //Emag the ticket machine to dispense burning tickets, as well as randomize its number to destroy the HoP's mind.
+	if(emagged)
+		return
+	to_chat(user, "<span class='warning'>You overload [src]'s bureaucratic logic circuitry to its MAXIMUM setting.</span>")
+	ticket_number = rand(0,max_number)
+	current_number = ticket_number
+	emagged = TRUE
+	if(tickets.len)
+		for(var/obj/item/ticket_machine_ticket/ticket in tickets)
+			ticket.audible_message("<span class='notice'>\the [ticket] disperses!</span>")
+			qdel(ticket)
+		tickets.Cut()
+	update_icon()
+
+/obj/machinery/ticket_machine/Initialize()
+	. = ..()
+	update_icon()
+
+/obj/machinery/ticket_machine/proc/increment()
+	if(current_number > ticket_number)
+		return
+	if(current_number && !(emagged) && tickets[current_number])
+		tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] disperses!</span>")
+		qdel(tickets[current_number])
+	if(current_number < ticket_number)
+		current_number ++ //Increment the one we're serving.
+		playsound(src, 'sound/misc/announce_dig.ogg', 50, FALSE)
+		atom_say("Now serving ticket #[current_number]!")
+		if(!(emagged) && tickets[current_number])
+			tickets[current_number].audible_message("<span class='notice'>\the [tickets[current_number]] vibrates!</span>")
+		update_icon() //Update our icon here rather than when they take a ticket to show the current ticket number being served
+
+/obj/machinery/ticket_machine_button
+	name = "increment ticket counter"
+	desc = "Use this button after you've served someone to tell the next person to come forward."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "crema_switch"
+	req_access = list()
+	var/id = 1
+	var/cooldown = FALSE
+
+/obj/machinery/ticket_machine_button/attack_ghost(mob/user)
+	if(user.can_advanced_admin_interact())
+		return attack_hand(user)
+
+/obj/machinery/ticket_machine_button/attack_hand(mob/user)
+	if(allowed(usr) || user.can_advanced_admin_interact())
+		for(var/obj/machinery/ticket_machine/M in world)
+			if(M.id == id)
+				if(cooldown)
+					return
+				cooldown = TRUE
+				M.increment()
+				addtimer(VARSET_CALLBACK(src, cooldown, FALSE), 10)
+	else
+		to_chat(usr, "<span class='warning'>Access denied.</span>")
+
+
+/obj/machinery/ticket_machine/update_icon()
+	switch(ticket_number) //Gives you an idea of how many tickets are left
+		if(0 to 49)
+			icon_state = "ticketmachine_100"
+		if(50 to 99)
+			icon_state = "ticketmachine_50"
+		if(100)
+			icon_state = "ticketmachine_0"
+	handle_maptext()
+
+/obj/machinery/ticket_machine/proc/handle_maptext()
+	switch(ticket_number) //This is here to handle maptext offsets so that the numbers align.
+		if(0 to 9)
+			maptext_x = 13
+		if(10 to 99)
+			maptext_x = 10
+		if(100)
+			maptext_x = 8
+	maptext = "[current_number]" //Finally, apply the maptext
+
+/obj/machinery/ticket_machine/attackby(obj/item/I, mob/user, params)
+	..()
```

Comment:
this should be under an `else` and `return ..()` otherwise you'll bap the machine with the hand labeler refill thing.

Reply:
Fixed